build.sh

```diff
@@ -0,0 +1,176 @@
+#!/bin/bash
+
+# Copyright (c) 2019, NVIDIA CORPORATION.
+
+# cuml build script
+
+# This script is used to build the component(s) in this repo from
+# source, and can be called with various options to customize the
+# build as needed (see the help output for details)
+
+# Abort script on first error
+set -e
+
+NUMARGS=$#
+ARGS=$*
+
+# NOTE: ensure all dir changes are relative to the location of this
+# script, and that this script resides in the repo dir!
+REPODIR=$(cd $(dirname $0); pwd)
+
+VALIDARGS="clean libcuml cuml prims -v -g -n --buildAllGPUArch --multigpu -h --help"
+HELP="$0 [<target> ...] [<flag> ...]
+ where <target> is:
+   clean             - remove all existing build artifacts and configuration (start over)
```

Comment:
This doesn't clean the faiss folder as far as I'm aware, which is probably beyond the scope of the PR but we can add it as a todo (something like `deep-clean` since most of the times you don't want to recreate the faiss artifacts)

Reply:
I filed issue #629 and added a TODO